src/main/java/seedu/task/logic/parser/EditParser.java

```diff
@@ -44,45 +40,57 @@
-                );
-            } catch (IllegalValueException ive) {
-                return new IncorrectCommand(ive.getMessage());
-            }
-        } else if (eventMatcher.matches()){
-            try {
-                return new EditEventCommand(
-                        Integer.parseInt(eventMatcher.group("index").trim()),
-                        isFieldToBeEdited(eventMatcher.group("newname")),
-                        eventMatcher.group("newname").replaceFirst("/name","").trim(),
-                        isFieldToBeEdited(eventMatcher.group("newdescription")),
-                        eventMatcher.group("newdescription").replaceFirst("/desc", "").trim(),
-                        isFieldToBeEdited(eventMatcher.group("newduration")),
-                        eventMatcher.group("newduration").replaceFirst("/from", "").trim()
-                );
-            } catch (IllegalValueException ive) {
-                return new IncorrectCommand(ive.getMessage());
+        
+        if (args.isEmpty()) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+        
+        ArgumentTokenizer argsTokenizer = new ArgumentTokenizer(descriptionPrefix, deadlinePrefix, 
+                durationStartPrefix, durationEndPrefix, taskPrefix, eventPrefix, namePrefix);
+       
+        argsTokenizer.tokenize(args.trim());
+        
+        try {           
+            Optional <String> taskIndex = argsTokenizer.getValue(taskPrefix);
+            Optional <String> eventIndex = argsTokenizer.getValue(eventPrefix);
+            Optional <String> name = argsTokenizer.getValue(namePrefix);
+            Optional <String> description = argsTokenizer.getValue(descriptionPrefix);
+            Optional <String> startDuration = argsTokenizer.getValue(durationStartPrefix);
+            Optional <String> endDuration = argsTokenizer.getValue(durationEndPrefix);
+            Optional <String> deadline = argsTokenizer.getValue(deadlinePrefix);
+            
+            if (eventIndex.isPresent()) { 
+                int index = getIndex(eventIndex.get().trim());
+                return new EditEventCommand(index, name.orElse(""), description.orElse(""), 
+                            startDuration.orElse(""), endDuration.orElse(""));                              
+            } else if (taskIndex.isPresent()) {  
+                int index = getIndex(taskIndex.get().trim());
+                return new EditTaskCommand(index, name.orElse(""), description.orElse(""), deadline.orElse(""));             
+            } else {
+                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
             }
-        }else {
+        } catch (IllegalValueException ive) {
+            System.out.println("IVE");
+            return new IncorrectCommand(ive.getMessage());
```

Comment:
Can remove the print statement, I used it for debugging previously. I've actually removed it in a later commit, so not sure why its appearing here.